src/index/IndexImpl.Text.cpp

```diff
@@ -202,12 +202,10 @@ void IndexImpl::buildDocsDB(const string& docsFileName) const {
     currentContextId++;
     currentOffset += static_cast<off_t>(lineView.size());
   }
-  offsets.push_back(currentOffset);
-
-  for (size_t i = 0; i < offsets.size(); ++i) {
-    off_t cur = offsets[i];
-    ofs.write(reinterpret_cast<const char*>(&cur), sizeof(cur));
-  }
+  ofs.write(reinterpret_cast<const char*>(offsets.data()),
+            sizeof(off_t) * offsets.size());
+  ofs.write(reinterpret_cast<const char*>(&currentOffset),
+            sizeof(currentOffset));
```

Comment:
I prefer a single write call, so please keep the last `push_back`. It is easier to read (part of te above algorithm, and then there comes a single write)

Reply:
Not saying that it makes a huge difference, but I changed it because I think it's silly to write something that's already in memory to disk (potentially triggering a reallocation), just to immediately retrieve it again.
But I definitely see your point